src/main/kotlin/com/openlattice/indexing/BackgroundIndexingService.kt

```diff
@@ -282,6 +304,38 @@ class BackgroundIndexingService(
 
     }
 
+    internal fun unindexEntities(
+            entitySet: EntitySet,
+            batchToIndex: Map<UUID, OffsetDateTime>,
+            markAsIndexed: Boolean = true
+    ): Int {
+        val esb = Stopwatch.createStarted()
+
+        val indexCount: Int
+
+        if (batchToIndex.isNotEmpty() &&
+                elasticsearchApi.deleteEntityDataBulk(entitySet.id, entitySet.entityTypeId, batchToIndex.keys)) {
+            indexCount = if (markAsIndexed) {
+                dataManager.markAsIndexed(mapOf(entitySet.id to batchToIndex), false)
+            } else {
+                batchToIndex.size
+            }
+
+            logger.info(
+                    "Un-indexed batch of {} elements for {} ({}) in {} ms",
+                    indexCount,
+                    entitySet.name,
+                    entitySet.id,
+                    esb.elapsed(TimeUnit.MILLISECONDS)
+            )
+        } else {
+            indexCount = 0
+            logger.error("Failed to un-index elements of entity set {} with entity key ids: {}", entitySet.id, batchToIndex.keys)
+
+        }
+        return indexCount
+    }
```

Comment:
Can invert to this for less nesting and (IMO) more readable code:
```
if ( batchToIndex.isEmpty() || !elasticsearchApi.deleteEntityDataBulk(entitySet.id, entitySet.entityTypeId, batchToIndex.keys) {
    logger.error("Failed. . . ")
    return 0
}
// Things currently inside of that first `If()`
```

Reply:
👍👍👍